nifi-nar-bundles/nifi-azure-bundle/nifi-azure-processors/src/main/java/org/apache/nifi/processors/azure/eventhub/ConsumeAzureEventHub.java

```diff
@@ -381,313 +366,285 @@ protected Collection<ValidationResult> customValidate(ValidationContext validati
     }
 
     @Override
-    public void onPropertyModified(PropertyDescriptor descriptor, String oldValue, String newValue) {
+    public void onPropertyModified(final PropertyDescriptor descriptor, final String oldValue, final String newValue) {
         if (RECORD_READER.equals(descriptor)) {
-            isRecordReaderSet = !StringUtils.isEmpty(newValue);
+            isRecordReaderSet = StringUtils.isNotEmpty(newValue);
         } else if (RECORD_WRITER.equals(descriptor)) {
-            isRecordWriterSet = !StringUtils.isEmpty(newValue);
-        }
-    }
-
-    public class EventProcessorFactory implements IEventProcessorFactory<EventProcessor> {
-        @Override
-        public EventProcessor createEventProcessor(PartitionContext context) throws Exception {
-            final EventProcessor eventProcessor = new EventProcessor();
-            return eventProcessor;
+            isRecordWriterSet = StringUtils.isNotEmpty(newValue);
         }
     }
 
-    public class EventProcessor implements IEventProcessor {
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSessionFactory sessionFactory) {
+        if (eventProcessorClient == null) {
+            eventProcessorClient = createClient(context);
+            eventProcessorClient.start();
 
-        @Override
-        public void onOpen(PartitionContext context) throws Exception {
-            getLogger().info("Consumer group {} opened partition {} of {}",
-                    new Object[]{context.getConsumerGroupName(), context.getPartitionId(), context.getEventHubPath()});
-        }
+            processSessionFactory = sessionFactory;
```

Comment:
I'd suggest setting `processSessionFactory` before starting the client because the client's event handler may try to use it before the variable gets initialized (almost zero chance though).